includes/embeds/class-amp-instagram-embed-handler.php

```diff
@@ -170,20 +172,34 @@ public function sanitize_raw_embeds( Document $dom ) {
 	 * @param DOMElement $node The DOMNode to adjust and replace.
 	 */
 	private function create_amp_instagram_and_replace_node( $dom, $node ) {
-		$instagram_id = $this->get_instagram_id_from_url( $node->getAttribute( 'data-instgrm-permalink' ) );
+		$permalink    = $node->getAttribute( 'data-instgrm-permalink' );
+		$instagram_id = $this->get_instagram_id_from_url( $permalink );
 
-		$node_args = [
-			'data-shortcode' => $instagram_id,
-			'layout'         => 'responsive',
-			'width'          => $this->DEFAULT_WIDTH,
-			'height'         => $this->DEFAULT_HEIGHT,
-		];
+		if ( $instagram_id ) {
+			$node_args = [
+				'data-shortcode' => $instagram_id,
+				'layout'         => 'responsive',
+				'width'          => $this->DEFAULT_WIDTH,
+				'height'         => $this->DEFAULT_HEIGHT,
+			];
 
-		if ( true === $node->hasAttribute( 'data-instgrm-captioned' ) ) {
-			$node_args['data-captioned'] = '';
-		}
+			if ( true === $node->hasAttribute( 'data-instgrm-captioned' ) ) {
+				$node_args['data-captioned'] = '';
+			}
 
-		$new_node = AMP_DOM_Utils::create_node( $dom, $this->amp_tag, $node_args );
+			$new_node = AMP_DOM_Utils::create_node( $dom, $this->amp_tag, $node_args );
+		} else {
+			$new_node = AMP_DOM_Utils::create_node(
+				$dom,
+				Tag::A,
+				[
+					Attribute::HREF   => esc_url_raw( $permalink ),
+					Attribute::CLASS_ => 'amp-wp-embed-fallback',
+				]
+			);
+
+			$new_node->textContent = esc_html( $permalink );
```

Comment:
No need to escape in a DOM context:

```suggestion
			$new_node->textContent = $permalink;
```

Reply:
Maybe it would be better to have link text like:

```suggestion
			$new_node->textContent = __( 'View on Instagram', 'amp' );
```